src/BuildingRegistry.Importer.Console/CommandGenerator.cs

```diff
@@ -258,12 +258,33 @@ public IEnumerable<dynamic> GenerateUpdateCommandsFor(int key, DateTime @from, D
 
             if (importMode == ImportMode.Update)
             {
-                var houseNumbersForUpdate = importTerrainObjectHouseNumberCommands
-                    .Where(x => x.Timestamp > from.ToCrabInstant() && x.Timestamp <= until.ToCrabInstant())
-                    .Select(x => x.HouseNumberId).ToList();
+                var importTerrainObjectHouseNumberFromCrabInRange = importTerrainObjectHouseNumberCommands
+                    .Where(x => x.Timestamp > @from.ToCrabInstant() && x.Timestamp <= until.ToCrabInstant())
+                    .ToList();
+
+                var houseNumbersForUpdate = importTerrainObjectHouseNumberFromCrabInRange.Select(x => x.HouseNumberId).ToList();
 
                 if (houseNumbersForUpdate.Any())
                 {
+                    var terrainObjectHouseNumberHouseNrCombinationsBeforeToday = importTerrainObjectHouseNumberCommands
+                        .Where(x => x.Timestamp <= from.ToCrabInstant())
+                        .Select(x => Tuple.Create(x.TerrainObjectHouseNumberId, x.HouseNumberId))
```

Comment:
using a Anonymous type will make the code far more readable then have to remember what item1 and item2 are

`.Select(x => new { x.TerrainObjectHouseNumberId, x.HouseNumberId })`